static/app/views/dashboards/widgetBuilder/components/visualize.tsx

```diff
@@ -0,0 +1,275 @@
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {
+  type AggregationKeyWithAlias,
+  generateFieldAsString,
+  parseFunction,
+  prettifyTagKey,
+} from 'sentry/utils/discover/fields';
+import useCustomMeasurements from 'sentry/utils/useCustomMeasurements';
+import useOrganization from 'sentry/utils/useOrganization';
+import useTags from 'sentry/utils/useTags';
+import {getDatasetConfig} from 'sentry/views/dashboards/datasetConfig/base';
+import {DisplayType, WidgetType} from 'sentry/views/dashboards/types';
+import {SectionHeader} from 'sentry/views/dashboards/widgetBuilder/components/common/sectionHeader';
+import {useWidgetBuilderContext} from 'sentry/views/dashboards/widgetBuilder/contexts/widgetBuilderContext';
+import {BuilderStateAction} from 'sentry/views/dashboards/widgetBuilder/hooks/useWidgetBuilderState';
+import {FieldValueKind} from 'sentry/views/discover/table/types';
+import {useSpanTags} from 'sentry/views/explore/contexts/spanTagsContext';
+
+function Visualize() {
+  const organization = useOrganization();
+  const {state, dispatch} = useWidgetBuilderContext();
+  let tags = useTags();
+  const {customMeasurements} = useCustomMeasurements();
+
+  const isChartWidget =
+    state.displayType !== DisplayType.TABLE &&
+    state.displayType !== DisplayType.BIG_NUMBER;
+  const numericSpanTags = useSpanTags('number');
+  const stringSpanTags = useSpanTags('string');
+  if (state.dataset === WidgetType.SPANS && isChartWidget) {
+    tags = numericSpanTags;
+  } else if (state.dataset === WidgetType.SPANS && !isChartWidget) {
+    tags = stringSpanTags;
+  }
+
+  const datasetConfig = useMemo(() => getDatasetConfig(state.dataset), [state.dataset]);
+
+  const fields = isChartWidget ? state.yAxis : state.fields;
+  const updateAction = isChartWidget
+    ? BuilderStateAction.SET_Y_AXIS
+    : BuilderStateAction.SET_FIELDS;
+
+  const fieldOptions = useMemo(
+    () => datasetConfig.getTableFieldOptions(organization, tags, customMeasurements),
+    [organization, tags, customMeasurements, datasetConfig]
+  );
+
+  // TODO: no parameters should show up as primary options?
+  // let aggregateOptions = Object.values(fieldOptions).filter(
+  //   option => option.value.kind === 'function' && option.value.meta.parameters.length > 0
+  // );
+  const aggregateOptions = useMemo(
+    () =>
+      Object.values(fieldOptions).filter(option =>
+        datasetConfig.filterYAxisOptions?.(state.displayType ?? DisplayType.TABLE)(option)
+      ),
+    [fieldOptions, state.displayType, datasetConfig]
+  );
+
+  // Used to extract selected aggregates and parameters from the fields
+  const stringFields = fields?.map(generateFieldAsString);
+
+  return (
+    <Fragment>
+      <SectionHeader
+        title={t('Visualize')}
+        tooltipText={t(
+          "This is the data you'd be visualizing in the display. If the overlay units conflict, the charts will always base it off of the first line."
```

Comment:
just a heads up, when I was talking to Vasudha about the tooltip texts she said we're taking some from explore and Visualize is one of them. She told me to use this when I asked her 

![image](https://github.com/user-attachments/assets/3d9a519b-076c-4d22-a501-5dd6878725de)